### DISCUSSION TITLE
migration failed while working with v1.9.1

### DISCUSSION THREAD

**Discussion by Amanb1145** [ory/hydra#3529] [Q&A]
Hey all, I am trying v1.9.1 on ubuntu 20.04 and while running the command `docker-compose -f quickstart.yml -f quickstart-postgres.yml up` I am getting the following error-

```
time=2023-05-30T17:25:43Z level=fatal msg=Could not ensure that signing keys for "hydra.openid.id-token" exists. If you are running against a persistent SQL database this is most likely because your "secrets.system" ("SECRETS_SYSTEM" environment variable) is not set or changed. When running with an SQL database backend you need to make sure that the secret is set and stays the same, unless when doing key rotation. This may also happen when you forget to run "hydra migrate sql".. audience=application error=map[message:failed to connect to `host=postgresd user=hydra database=hydra`: hostname resolving error (lookup postgresd on 127.0.0.11:53: server misbehaving)] service_name=ORY Hydra service_version=v1.9.0-rc.0
I have used this before but this is the first time I have encountered this issue
```
any ideas ?
Thanks

**Comment by vinckr** [comment]
Hello @Amanb1145 

I would suggest to use the latest version of Ory Hydra, that should run without any issues on the docker-compose quickstart.


The error message you posted suggests a couple of potential issues:
- The SECRETS_SYSTEM environment variable is not set or has changed. This variable is crucial when running Ory Hydra with a persistent SQL database backend. You need to ensure that the secret is set and remains the same, except when performing key rotation.
- The database migration might not have been run. The `hydra migrate sql` command is necessary to create the database schema and run database migrations in case of an upgrade.
You can run the migration command in Docker using the following command:

```
docker-compose -f quickstart.yml -f quickstart-postgres.yml exec hydra \
    hydra migrate sql $DATABASE_URL
```

Please replace $DATABASE_URL with your actual database connection string.
- Also, the error message indicates a failure to connect to the database. Please ensure that your database is running and accessible. If you're using Docker Compose, the database service should be defined in your docker-compose.yml file and should be running without issues.
